managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1581,7 +1581,7 @@ synchronized void ledgerClosed(final LedgerHandle lh) {
 
         maybeOffloadInBackground(NULL_OFFLOAD_PROMISE);
 
-        if (!pendingAddEntries.isEmpty()) {
+        if (!pendingAddEntries.isEmpty() && isNeededCreateNewLedgerAfterCloseLedger()) {
             // Need to create a new ledger to write pending entries
```

Comment:
It looks like duplicate code here

Reply:
it is neccessary here

Reply:
It is necessary, but the code is duplicate  with createLedgerAfterClosed(). Can we call createLedgerAfterClosed directly

Reply:
ok, i got it, i will change it later.